aqt/metadata.py

```diff
@@ -591,16 +595,14 @@ def fetch_arches(self, version: Version) -> List[str]:
     def fetch_versions(self, extension: str = "") -> Versions:
         def filter_by(ver_ext: Tuple[Optional[Version], str]) -> bool:
             version, ext = ver_ext
-            return version and (self.spec is None or version in self.spec) and (ext == extension)
-
-        def get_version(ver_ext: Tuple[Version, str]):
-            return ver_ext[0]
+            return version is not None and (self.spec is None or version in self.spec) and (ext == extension)
 
         versions_extensions = self.get_versions_extensions(
             self.fetch_http(self.archive_id.to_url(), False), self.archive_id.category
         )
-        versions = sorted(filter(None, map(get_version, filter(filter_by, versions_extensions))))
-        iterables = itertools.groupby(versions, lambda version: version.minor)
+        opt_versions = map(lambda _tuple: _tuple[0], filter(filter_by, versions_extensions))
```

Comment:
snyk analyzer warns "Combine filter then map into one comprehension" can be "[x for x in y if z]" style.
```suggestion
        opt_versions = [_tuple[0] for _tuple in filter(filter_by, versions_extensions)]
```